src/spares/oc_presets

```diff
@@ -0,0 +1,208 @@
+// This file is part of OpenCollar.
+// Copyright (c) 2019 Romka Swallowtail
+// Licensed under the GPLv2.  See LICENSE for full details.
+
+string g_sSubMenu = "SizePresets";
+string g_sParentMenu = "Apps";
+
+string RESTORE = "RESTORE";
+string SAVE = "SAVE";
+string DEL = "DELETE";
+string NEW = "NEW";
+string UPMENU = "BACK";
+string g_sToken = "presets";
+
+//MESSAGE MAP
+integer CMD_OWNER = 500;
+integer CMD_WEARER = 503;
+
+integer NOTIFY = 1002;
+
+integer LM_SETTING_SAVE = 2000;
+//integer LM_SETTING_REQUEST = 2001;
+integer LM_SETTING_RESPONSE = 2002;
+integer LM_SETTING_DELETE = 2003;
+//integer LM_SETTING_EMPTY = 2004;
+integer REBOOT      = -1000;
+integer LINK_DIALOG = 3;
+integer LINK_SAVE   = 5;
+integer LINK_UPDATE = -10;
+
+integer MENUNAME_REQUEST = 3000;
+integer MENUNAME_RESPONSE = 3001;
+integer MENUNAME_REMOVE = 3003;
+
+integer DIALOG = -9000;
+integer DIALOG_RESPONSE = -9001;
+integer DIALOG_TIMEOUT = -9002;
+
+key g_kWearer;
+
+list g_lMenuIDs;
+integer g_iMenuStride = 3;
+
+list g_lPresets;  // [sName, "vScale/vPos/vRot"]
+
+Dialog(key kRCPT, string sPrompt, list lChoices, list lUtilityButtons, integer iPage, integer iAuth, string sMenuType) {
+    key kMenuID = llGenerateKey();
+    llMessageLinked(LINK_DIALOG, DIALOG, (string)kRCPT + "|" + sPrompt + "|" + (string)iPage + "|" + llDumpList2String(lChoices, "`") + "|" + llDumpList2String(lUtilityButtons, "`") + "|" + (string)iAuth, kMenuID);
+    integer iIndex = llListFindList(g_lMenuIDs, [kRCPT]);
+    if (~iIndex) g_lMenuIDs = llListReplaceList(g_lMenuIDs, [kRCPT, kMenuID, sMenuType], iIndex, iIndex + g_iMenuStride - 1);
+    else g_lMenuIDs += [kRCPT, kMenuID, sMenuType];
+}
+
+ForceUpdate() {
+    //workaround for https://jira.secondlife.com/browse/VWR-1168
+    llSetText(".", <1,1,1>, 1.0);
+    llSetText("", <1,1,1>, 1.0);
+}
+
+DoMenu(key kAv, integer iAuth) {
+    list lMyButtons ;
+    string sPrompt = "\nYou can save/restore size, position and rotation of %DEVICETYPE%.";
+
+    if (g_lPresets != []) lMyButtons += [RESTORE, SAVE, DEL];
+    else lMyButtons += ["-", SAVE, "-"];
+    Dialog(kAv, sPrompt, lMyButtons, [UPMENU], 0, iAuth, g_sSubMenu);
+}
+
+PresetMenu(key kAv, integer iAuth, string sType) {
+    string sPrompt;
+    list lMyButtons = llList2ListStrided(g_lPresets,0,-1,2);
+    list lUtils = [UPMENU];
+    if (sType == RESTORE) sPrompt = "\nSelect preset to restore size/position/rotation";
+    else if (sType == SAVE) {
+        if (g_lPresets == []) sType = NEW;
+        else {
+            sPrompt = "\nSelect preset or create new to save size/position/rotation";
+            lMyButtons += [NEW];
+        }
+    } else if (sType == DEL) sPrompt = "\nSelect preset or create new to delete size/position/rotation";
+    if (sType == NEW) {
+        sPrompt = "\nType new preset name";
+        lMyButtons = [];
+        lUtils = [];
+    }
+    Dialog(kAv, sPrompt, lMyButtons, lUtils, 0, iAuth, sType);
+}
+
+Restore(string sName, key kAv) {
```

Comment:
This appears to do nothing when I attempted to restore after having resized the collar and moved several of the prims around. Various other configurations did not result in restore working.

Reply:
did You do SAVE before?
Collar menu -> Apps -> SizePresets -> SAVE ,  type preset name in textbox and press Submit button to save current Collar size/position/rotation
after that, the saved preset will appear in the RESTORE menu

You can change the size and position of the collar through the Settings menu, and then save it as a new preset

Reply:
See my comment about what steps I took. I saved three different presets